packages/paste-core/components/chat-log/src/ChatAttachment.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Anchor} from '@twilio-paste/anchor';
+import {Button} from '@twilio-paste/button';
+import {Box} from '@twilio-paste/box';
+import {CloseIcon} from '@twilio-paste/icons/esm/CloseIcon';
+import {MediaObject, MediaFigure, MediaBody} from '@twilio-paste/media-object';
+import {Text} from '@twilio-paste/text';
+import {Truncate} from '@twilio-paste/truncate';
+
+import type {
+  ChatAttachmentProps,
+  ChatAttachmentContainerProps,
+  ChatAttachmentLinkProps,
+  ChatAttachmentDescriptionProps,
+} from './types';
+
+const ChatAttachmentLink = React.forwardRef<HTMLElement, ChatAttachmentLinkProps>(
+  ({children, href, element = 'CHAT_ATTACHMENT_LINK', ...props}, ref) => {
+    return (
+      <Anchor href={href} ref={ref} element={element} {...props}>
+        <Truncate title={children}>{children}</Truncate>
+      </Anchor>
+    );
+  }
+);
+
+ChatAttachmentLink.displayName = 'ChatAttachmentLink';
+
+const ChatAttachmentDescription = React.forwardRef<HTMLElement, ChatAttachmentDescriptionProps>(
+  ({children, element = 'CHAT_ATTACHMENT_DESCRIPTION', ...props}, ref) => {
+    return (
+      <Text
+        element={element}
+        ref={ref}
+        as="div"
+        fontSize="fontSize20"
+        color="colorTextWeaker"
+        lineHeight="lineHeight10"
+        {...props}
+      >
+        {children}
+      </Text>
+    );
+  }
+);
+
+ChatAttachmentDescription.displayName = 'ChatAttachmentDescription';
+
+const ChatAttachment = React.forwardRef<HTMLDivElement, ChatAttachmentProps>(
+  ({children, element = 'CHAT_ATTACHMENT', attachmentIcon: AttachmentIcon, ...props}, ref) => {
+    return (
```

Comment:
Suggestion: one other thing the ChatBubble/ChatMessageMeta do that is helpful for custom styles is pass the variant ('inbound'/'outbound') to the box.

Because the ChatAttachment component isn't always in the MessageVariantContext, I wonder if the component can check if the context exists and if it does pass that to the Box. 

(example here: https://github.com/twilio-labs/paste/blob/eb490746cb957a349fb217c13b886db4a7277342/packages/paste-core/components/chat-log/src/ChatBubble.tsx#L11)

Reply:
ohhh this is a good point and could def help someone doing customization. I'll work on this tomorrow- thanks @shleewhite!

Reply:
Okay, this is harder to implement here than in ChatBubble bc Attachment uses MediaObject as opposed to Box, so I'm going to leave this out, especially since it's style customizable w/ `element`.